### PR TITLE
SONARIAC-795 [Microsoft.Web/*] properties/siteConfig/publicNetworkAccess

### PR DESCRIPTION
**Pull request by mstachniuk**

### CODE REVIEW

File: iac-extensions/arm/src/main/java/org/sonar/iac/arm/checks/PublicNetworkAccessCheck.java [hunk 1]
```diff
@@ -132,6 +132,11 @@ public class PublicNetworkAccessCheck extends AbstractArmResourceCheck {
     "Microsoft.DocumentDB/mongoClusters/firewallRules",
     "Microsoft.Sql/servers/firewallRules",
     "Microsoft.Synapse/workspaces/firewallRules");
+
+  private static final List<String> PUBLIC_NETWORK_ACCESS_SIMPLIFIED_IN_SITE_CONFIG = List.of(
```
Comment (jonas-wielage-sonarsource) [inline iac-extensions/arm/src/main/java/org/sonar/iac/arm/checks/PublicNetworkAccessCheck.java:136]: Why is this also `SIMPLIFIED`? Shouldn't this only be `PUBLIC_NETWORK_ACCESS_SIMPLIFIED_IN_SITE_CONFIG `? Same for all related methods and test files

File: iac-extensions/arm/src/test/java/org/sonar/iac/arm/checks/PublicNetworkAccessCheckTest.java [hunk 2]
```diff
@@ -236,12 +236,31 @@ void shouldCheckDbForMySqlFlexibleServers() {
...
   }
 
+  static Stream<String> shouldCheckPublicNetworkAccessSimplifiedInSiteConfig() {
+    return Stream.of(
+      "Microsoft.Web/sites",
+      "Microsoft.Web/sites/slots");
+  }
+
+  @MethodSource
+  @ParameterizedTest(name = "[${index}] should check range public IP Address for type {0}")
```
Comment (jonas-wielage-sonarsource) [inline iac-extensions/arm/src/test/java/org/sonar/iac/arm/checks/PublicNetworkAccessCheckTest.java:252]: I didn't know that you could define custom names for the tests, looks quite nice!

**Review by jonas-wielage-sonarsource — APPROVED** [review]
LGTM! See my comments before merging

File: iac-extensions/arm/src/main/java/org/sonar/iac/arm/checks/PublicNetworkAccessCheck.java (hunk 1 above)
Comment (mstachniuk) [reply iac-extensions/arm/src/main/java/org/sonar/iac/arm/checks/PublicNetworkAccessCheck.java:136]: The `publicNetworkAccess` property in some cases can have the following sensitive values: `"Enabled", "EnabledForSessionHostsOnly", or "EnabledForClientsOnly" ` and in other cases only `Enabled` is sensitive. I call the second case `SIMPLIFIED`. If you have a better name for it, I'm happy to refactor it.

File: iac-extensions/arm/src/test/java/org/sonar/iac/arm/checks/PublicNetworkAccessCheckTest.java (hunk 2 above)
Comment (mstachniuk) [reply iac-extensions/arm/src/test/java/org/sonar/iac/arm/checks/PublicNetworkAccessCheckTest.java:252]: It's very useful for Parameterized Test

File: iac-extensions/arm/src/main/java/org/sonar/iac/arm/checks/PublicNetworkAccessCheck.java (hunk 1 above)
Comment (jonas-wielage-sonarsource) [reply iac-extensions/arm/src/main/java/org/sonar/iac/arm/checks/PublicNetworkAccessCheck.java:136]: Ah ok, I thought it had something to do with how deep the `enabled` is inside the properties.

**Pull request merged**